server/modules/resourceFactories/DynamoTableResource.js

```diff
@@ -91,6 +91,28 @@ class DynamoTableResource {
     });
   }
 
+  query(params) {
+    let self = this;
+    let request = {
+      TableName: this._tableName,
+      KeyConditionExpression: `${this._keyName} = :key`,
+      ExpressionAttributeValues: {
+        ":key": params.key
+      }
+    };
+
+    return this.client.query(request).promise().then(data => {
+      if (!data.Items) {
```

Comment:
Again data.Items.length would fail if Items is undefined. The existing code is much safer.. Could use _.isArray though.